rasa/core/actions/flows.py

```diff
@@ -0,0 +1,293 @@
+from typing import Text, List, Optional, Union, Any, Dict
+import logging
+from pypred import Predicate
+from rasa.core.actions import action
+from rasa.core.actions.forms import FormAction
+from rasa.core.channels import OutputChannel
+
+from rasa.shared.core.constants import (
+    ACTION_LISTEN_NAME,
+    REQUESTED_SLOT,
+    NEXT_STEP,
+)
+from rasa.shared.core.domain import Domain
+from rasa.shared.core.slots import TextSlot
+from rasa.shared.core.events import (
+    Event,
+    SlotSet,
+    FollowupAction,
+    ActiveLoop,
+    ActionExecutionRejected,
+)
+from rasa.core.nlg import NaturalLanguageGenerator
+from rasa.shared.core.trackers import DialogueStateTracker
+from rasa.utils.endpoints import EndpointConfig
+
+logger = logging.getLogger(__name__)
+
+
+class FlowAction(FormAction):
```

Comment:
before we go and implement more in this direction we should check if an action actually is the right place. 

my first guess would have been that this should be a policy (like the rule policy) tbh - but I am not so sure about this. let me open a thread in the engine channel to check what others think

Reply:
(might also something that we need to toy with)